pkg/issuer/acme/setup.go

```diff
@@ -167,6 +169,27 @@ func (a *Acme) Setup(ctx context.Context) error {
 	// registerAccount will also verify the account exists if it already
 	// exists.
 	account, err := a.registerAccount(ctx, cl)
+
+	// if we got an account successfully, we must check if the registered
+	// email is the same as in the issuer spec
+	registeredEmail := ""
+	if err == nil {
```

Comment:
Instead of wrapping this block in `if err == nil`, can we move it to *after* the `err != nil` check and thus assume that it *is* nil? We can then throw a distinct warning that updating the account failed if `err != nil` from: `account, err = cl.UpdateAccount(ctx, account)`.

Reply:
I wrote it before the `err != nil` block as it seemed that we would probably want to handle any errors from updating in a similar fashion to any from registering.
Should errors from updating just log a message and return the error, or should it behave as before but with all instances of `verify` changed to `update`?

Reply:
Yeah I think we should duplicate the logic and update the messages.

We can also log an event to say we are updating the email address here too, which will help users be sure that the email address has been updated.

Reply:
We should also keep the existing error code checking logic to handle known ACME error codes and skip retrying, to save DoSing the ACME API.